osidb/models.py

```diff
@@ -1595,6 +1596,160 @@ def bzsync(self, *args, bz_api_key, **kwargs):
+
+
+class AffectCVSSManager(ACLMixinManager, TrackingMixinManager):
+    """Affect CVSS Manager"""
+
+    @staticmethod
+    def create_cvss(affect, issuer, version, **extra_fields):
+        """return a new CVSS or update an existing CVSS without saving"""
+
+        try:
+            cvss = AffectCVSS.objects.get(affect=affect, issuer=issuer, version=version)
+            for attr, value in extra_fields.items():
+                setattr(cvss, attr, value)
+            return cvss
+
+        except ObjectDoesNotExist:
+            return AffectCVSS(
+                affect=affect, issuer=issuer, version=version, **extra_fields
+            )
+
+
+class CVSS(
+    ACLMixin,
+    AlertMixin,
+    TrackingMixin,
+):
+    """Model CVSS"""
+
+    class CVSSVersion(models.TextChoices):
+        VERSION2 = "V2", "version 2"
+        VERSION3 = "V3", "version 3"
+
+    class CVSSIssuer(models.TextChoices):
+        REDHAT = "RH", "Red Hat"
+        NIST = "NIST", "NIST"
+
+    CVSS_HANDLES = {CVSSVersion.VERSION2: CVSS2, CVSSVersion.VERSION3: CVSS3}
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    vector = models.CharField(max_length=100, blank=False)
+
+    version = models.CharField(choices=CVSSVersion.choices, max_length=10)
+
+    issuer = models.CharField(choices=CVSSIssuer.choices, max_length=16)
+
+    comment = models.TextField(blank=True)
+
+    # populated by the pre_save signal
+    score = models.FloatField(default=0)
+
+    def __str__(self):
+        return f"{self.score}/{self.vector}"
+
+    @property
+    def full_version(self):
+        """Full name of the CVSS version."""
+        return f"CVSS{self.version[1:]}"
+
+    @property
+    def cvss_handle(self):
+        """
+        Corresponding CVSS handle from CVSS library based on the version.
+        """
+        return self.CVSS_HANDLES[self.version]
+
+    @property
+    def cvss_object(self):
+        """
+        CVSS object from CVSS library parsed from the vector.
+        """
+        return self.cvss_handle(self.vector)
+
+    def _validate_cvss_string(self):
+        """
+        Use the cvss library to validate the CVSS vector string.
+        """
+        try:
+            self.cvss_handle(self.vector)
+        except CVSSError as e:
+            raise ValidationError(
+                f"Invalid CVSS: Malformed {self.full_version} string: {e}"
+            )
+
+    def _validate_cvss_comment(self):
+        """
+        For non-Red-Hat-issued CVSSs, the comment attribute should be blank.
+        """
+        if self.comment and self.issuer != self.CVSSIssuer.REDHAT:
+            raise ValidationError(
+                "CVSS comment can be set only for CVSSs issued by Red Hat."
+            )
+
+    def __eq__(self, o):
+        if isinstance(o, CVSS) and self.version == o.version:
+            return self.cvss_object == o.cvss_object
+        return False
```

Comment:
Unless you have a strong reason for this I'd rather not override `__eq__`, this also makes CVSS model object instances unhasheable.